contracts/SignatureValidatorManager.sol

```diff
@@ -0,0 +1,211 @@
+     *                  0x04 to 0x24 - 32 bytes hash of the signed message
+     *                  0x24 to 0x44 - 32 bytes offset to the start of `bytes` parameter
+     *                  0x44 to 0x64 - 32 bytes length of `bytes` parameter
+     *                  0x64 to 0x68 - 4 bytes of Signature selector
+     *                  0x68 to 0x88 - 32 bytes domain separator
+     *                  0x88 to 0xa8 - 32 bytes struct hash
+     *                  0xa8 to end - contains offset, length of bytes, and actual bytes containing signatures
+     *             Else:
+     *                 bytes containing signature data
+     *                 default validation flow will be used which will depend on the account implementation
+     *
+     */
+    function handle(address account, address sender, uint256 /* value */, bytes calldata data) external override returns (bytes memory) {
+        // Skip first 4 bytes of data as it contains function selector
+        (bytes32 messageHash, bytes memory signatureData) = abi.decode(data[0x4:], (bytes32, bytes));
+
+        address signatureValidatorHooksAddress = signatureValidatorHooks[account];
+        bytes memory prevalidationData;
+        bytes memory returnData;
+
+        if (signatureValidatorHooksAddress != address(0)) {
+            checkPermittedModule(signatureValidatorHooksAddress, MODULE_TYPE_SIGNATURE_VALIDATOR_HOOKS);
+            prevalidationData = ISafeProtocolSignatureValidatorHooks(signatureValidatorHooksAddress).preValidationHook(
+                account,
+                sender,
+                data
+            );
+        }
+
+        if (bytes4(data[0x64:0x68]) == SIGNATURE_VALIDATOR_SELECTOR) {
+            returnData = abi.encode(validateWithSignatureValdiator(account, sender, messageHash, data[0x68:]));
+
+            if (signatureValidatorHooksAddress != address(0)) {
+                ISafeProtocolSignatureValidatorHooks(signatureValidatorHooksAddress).postValidationHook(account, prevalidationData);
+            }
+            return returnData;
+        }
+
+        returnData = defaultValidator(account, messageHash, signatureData);
+        if (signatureValidatorHooksAddress != address(0)) {
+            ISafeProtocolSignatureValidatorHooks(signatureValidatorHooksAddress).postValidationHook(account, prevalidationData);
+        }
+        return returnData;
+    }
+
+    /**
+     * @notice A view function for default signature validation flow.
+     * @param account Address of the account whose signature is to be validated. Account should support function `checkSignatures(bytes32, bytes, bytes)`
+     * @param hash bytes32 hash of the data that is signed
+     * @param signatures Arbitrary length bytes array containing the signatures
+     */
+    function defaultValidator(address account, bytes32 hash, bytes memory signatures) internal view returns (bytes memory) {
+        ISafeAccount(account).checkSignatures(hash, "", signatures);
```

Comment:
Since we're trying to be account-agnostic I'd propose renaming `ISafeAccount` to `IAccount`

Reply:
Introduced `ISafeAccount` here because defaultValidation would be account dependent and `checkSignatures(bytes32,bytes,bytes)` is specific to Safe accounts.

Do you want to say that we add this function in `IAccount`?

Reply:
I think in the current form, it makes it an account requirement if they want to use signature validators.

Reply:
> I think in the current form, it makes it an account requirement if they want to use signature validators.

Its an account requirement only for default validation flow.

Reply:
https://github.com/safe-global/safe-core-protocol/pull/109/commits/f7dad8ccc32ce3bdad6b451fc1acaaf325714446